vitest.sh

```diff
@@ -0,0 +1,116 @@
+#set -x
+# Returns the environment variables needed for a test run based on dialect env var
+# and version env var. Then runs vitest with the provided arguments.
+
+# Usage: ./vitest.sh <dialect> <version> [args]
+# Example: ./vitest.sh postgres 13 --coverage
+
+# postgres versions: 13, 14, 15, 16
+# mysql versions: 8
+# mariadb versions: 10
+
+test() {
+  dialect=$1
+  version=$2
+  args=$3
+  echo $args
+
+  export NODE_OPTIONS='--no-warnings'
+  export DEBUG='1'
+
+  if [ "$dialect" = "all" ]; then
+    args="--reporter=dot $2 $args"
+    docker compose up -d --wait
+    setenv "mysql" 8 && pnpm vitest run $args
+    setenv "mariadb" 10 && pnpm vitest run $args
+    setenv "postgres" 13 && pnpm vitest run $args
+    setenv "postgres" 14 && pnpm vitest run $args
+    setenv "postgres" 15 && pnpm vitest run $args
+    setenv "postgres" 16 && pnpm vitest run $args
+    docker compose down
+  else
+    setenv "$dialect" "$version"
+
+    service="${dialect}${version}"
+    docker compose up "${service}" -d --wait "${service}"
+
+    pnpm vitest "$args"
+    docker compose down "${service}"
+  fi
+}
+
+setenv() {
+  dialect=$1
+  version=$2
+
+  export SERVICE="${dialect}${version}"
+
+  case $dialect in
+  postgres)
+    export DIALECT=postgresql
+    export DB_HOST=localhost
+    export DB_USER=postgres_user
+    export DB_PASSWORD=postgres_password
+    export DB_NAME=test_database
+    export DB_SCHEMA=postgres_schema
+    export DB_DATABASE=test_database
+
+    case $version in
+    13)
+      export DB_PORT=5432
+      ;;
+    14)
+      export DB_PORT=5433
+      ;;
+    15)
+      export DB_PORT=5434
+      ;;
+    16)
+      export DB_PORT=5435
+      ;;
+    *)
+      echo "Unknown version: $version. Options: 13, 14, 15, 16"
+      exit 1
+      ;;
+    esac
+    ;;
+  mysql)
+    export DIALECT=mysql2
+    export DB_DATABASE=test_database
+    export DB_USER=root
+    export DB_PASSWORD=mysql_root_password
+    export DB_HOST=localhost
+    case $version in
+    8)
+      export DB_PORT=3306
+      ;;
+    *)
+      echo "Unknown version: $version. Options: 8"
+      exit 1
+      ;;
+    esac
+    ;;
+  mariadb)
+    export DIALECT=mysql2
+    export DB_DATABASE=test_database
+    export DB_USER=root
+    export DB_PASSWORD=mariadb_root_password
+    export DB_HOST=localhost
+    case $version in
+    10)
+      export DB_PORT=3300
+      ;;
+    *)
+      echo "Unknown version: $version. Options: 10"
+      exit 1
+      ;;
+    esac
+    ;;
+  *)
+    echo "Unknown dialect: $dialect. Options: postgres, mysql, mariadb"
+    exit 1
+    ;;
+  esac
+}
+
+test "$1" "$2" "${@:3}"
```

Comment:
shell is usually four spaces. This is difficult to read. 

shellcheck may be worth adding to the linter.

Reply:
Updated to 4 spaces instead of 2. I added shellcheck as a package script but it's not being run in CI for the moment.